Mathlib/RingTheory/Nullstellensatz.lean

```diff
@@ -27,164 +27,160 @@ noncomputable section
+theorem vanishingIdeal_empty : vanishingIdeal k (∅ : Set (σ → K)) = ⊤ :=
   le_antisymm le_top fun _ _ x hx => absurd hx (Set.notMem_empty x)
 
 theorem le_vanishingIdeal_zeroLocus (I : Ideal (MvPolynomial σ k)) :
-    I ≤ vanishingIdeal (zeroLocus I) := fun p hp _ hx => hx p hp
+    I ≤ vanishingIdeal k (zeroLocus K I) := fun p hp _ hx => hx p hp
 
-theorem zeroLocus_vanishingIdeal_le (V : Set (σ → k)) : V ≤ zeroLocus (vanishingIdeal V) :=
+theorem zeroLocus_vanishingIdeal_le (V : Set (σ → K)) : V ≤ zeroLocus K (vanishingIdeal k V) :=
   fun V hV _ hp => hp V hV
 
 theorem zeroLocus_vanishingIdeal_galoisConnection :
-    @GaloisConnection (Ideal (MvPolynomial σ k)) (Set (σ → k))ᵒᵈ _ _ zeroLocus vanishingIdeal :=
+    @GaloisConnection (Ideal (MvPolynomial σ k)) (Set (σ → K))ᵒᵈ _ _
+      (zeroLocus K) (vanishingIdeal k) :=
   GaloisConnection.monotone_intro (fun _ _ ↦ vanishingIdeal_anti_mono)
     (fun _ _ ↦ zeroLocus_anti_mono) le_vanishingIdeal_zeroLocus zeroLocus_vanishingIdeal_le
 
-theorem le_zeroLocus_iff_le_vanishingIdeal {V : Set (σ → k)} {I : Ideal (MvPolynomial σ k)} :
-    V ≤ zeroLocus I ↔ I ≤ vanishingIdeal V :=
+theorem le_zeroLocus_iff_le_vanishingIdeal {V : Set (σ → K)} {I : Ideal (MvPolynomial σ k)} :
+    V ≤ zeroLocus K I ↔ I ≤ vanishingIdeal k V :=
   zeroLocus_vanishingIdeal_galoisConnection.le_iff_le
 
 theorem zeroLocus_span (S : Set (MvPolynomial σ k)) :
-    zeroLocus (Ideal.span S) = { x | ∀ p ∈ S, eval x p = 0 } :=
+    zeroLocus K (Ideal.span S) = { x | ∀ p ∈ S, aeval x p = 0 } :=
   eq_of_forall_le_iff fun _ => le_zeroLocus_iff_le_vanishingIdeal.trans <|
     Ideal.span_le.trans forall₂_swap
 
-theorem mem_vanishingIdeal_singleton_iff (x : σ → k) (p : MvPolynomial σ k) :
-    p ∈ (vanishingIdeal {x} : Ideal (MvPolynomial σ k)) ↔ eval x p = 0 :=
+theorem mem_vanishingIdeal_singleton_iff (x : σ → K) (p : MvPolynomial σ k) :
+    p ∈ (vanishingIdeal k {x} : Ideal (MvPolynomial σ k)) ↔ aeval x p = 0 :=
   ⟨fun h => h x rfl, fun hpx _ hy => hy.symm ▸ hpx⟩
 
-instance vanishingIdeal_singleton_isMaximal {x : σ → k} :
-    (vanishingIdeal {x} : Ideal (MvPolynomial σ k)).IsMaximal := by
-  have : Function.Bijective
-      (Ideal.Quotient.lift _ (eval x) fun p h ↦ (mem_vanishingIdeal_singleton_iff x p).mp h) := by
-    refine ⟨(injective_iff_map_eq_zero _).mpr fun p hp ↦ ?_, fun z ↦
-      ⟨(Ideal.Quotient.mk (vanishingIdeal {x} : Ideal (MvPolynomial σ k))) (C z), by simp⟩⟩
-    obtain ⟨q, rfl⟩ := Ideal.Quotient.mk_surjective p
-    rwa [Ideal.Quotient.lift_mk, ← mem_vanishingIdeal_singleton_iff,
-      ← Quotient.eq_zero_iff_mem] at hp
-  rw [← bot_quotient_isMaximal_iff, isMaximal_iff_of_bijective _ this]
-  exact bot_isMaximal
+instance vanishingIdeal_singleton_isPrime {x : σ → K} :
+    (vanishingIdeal k {x} : Ideal (MvPolynomial σ k)).IsPrime := by
+  convert RingHom.ker_isPrime (aeval (R := k) x)
+  ext; simp
+
+instance vanishingIdeal_singleton_isMaximal {x : σ → K} :
```

Comment:
There's no need to name this explicitly but if you wish to do so then it should be something like:
```suggestion
instance instIsMaximalVanishingIdealSingleton {x : σ → K} :
```